test/Sema/diag_unintended_optional_behavior.swift

```diff
@@ -82,34 +82,33 @@ func warnNestedOptionalToOptionalAnyCoercion(_ a: Int?, _ b: Any??, _ c: Int???,
   takesOptionalAny(c as Any?, d as Any?)
 }
 
-func warnIUOToAnyCoercion(_ a: Int!, _ b: Any?!) {
-  _ = takeAny(a, b) // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}{{16-16= ?? <#default value#>}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}{{16-16=!}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{16-16= as Any}}
-  // expected-warning@-4 {{expression implicitly coerced from 'Any??' to 'Any'}}
-  // expected-note@-5 {{force-unwrap the value to avoid this warning}}{{19-19=!!}}
-  // expected-note@-6 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{19-19= as Any}}
+class C {
+  var a: Int!
+  var b: Any?!
+  func returningIUO() -> Int! { return a }
+  func returningAny() -> Any { return a }
```

Comment:
For completeness can you test a subscript and an initializer too?

Reply:
Good call, especially since I forgot to add subscript support!

I had made a note of doing so but then never came back and actually did it in the original commit.